packages/mock-block-dock/package.json

```
@@ -24,12 +24,16 @@
     "dist"
   ],
   "scripts": {
-    "build": "yarn clean && yarn build:cjs && yarn build:esm",
-    "build:cjs": "tsc --p tsconfig.build.json --module commonjs --outDir dist/cjs ",
+    "build": "yarn clean && yarn generate-version-file && yarn build:cjs && yarn build:esm && yarn copy-files",
+    "build:cjs": "tsc --p tsconfig.build.json --module commonjs --outDir dist/cjs",
     "build:dev": "yarn clean && cross-env NODE_ENV=development webpack --mode development",
     "build:esm": "tsc --p tsconfig.build.json",
     "clean": "rimraf ./dist/",
-    "dev": "concurrently -n webpack,webpack-dev-server -c green,cyan \"yarn build:dev --watch\" \"yarn run-dev-server\"",
+    "copy-files": "yarn copy-files:cjs && yarn copy-files:esm",
+    "copy-files:cjs": "copyfiles -u 1 ./src/assets/* dist/cjs/",
+    "copy-files:esm": "copyfiles -u 1 ./src/assets/* dist/esm/",
+    "dev": "yarn generate-version-file && concurrently -n webpack,webpack-dev-server -c green,cyan \"yarn build:dev --watch\" \"yarn run-dev-server\"",
+    "generate-version-file": "node -p \"'export const MOCK_BLOCK_DOCK_VERSION = ' + JSON.stringify(require('./package.json').version) + ';'\" > src/version.ts",
```

Optional name change: 

```suggestion
    "codegen": "node -p \"'export const MOCK_BLOCK_DOCK_VERSION = ' + JSON.stringiy(require('./package.json').version) + ';'\" > src/version.ts",
```

We already have `codegen` script in `site` and it is launched via turbo. If you look at `turbo.json`, you'll see that `build` already depends on `codegen`. This means that you won't have to include `yarn generate-version-file` in `build`. We can create the same  relationship between `codegen` and `dev` too.

---

Nice!

---

I will address in #555

---

Renamed to codegen in 5f513df16fbd7a5e7c4018ef805